### PR TITLE
Draft to add text-generation parameters

### PR DESCRIPTION
**Pull request by Wauplin**
This PR adds pages for `text-generation` and `chat-completion` pages. The `generate.ts` code is getting uglier and uglier with all the cases to handle but hopefully we are seeing the end of it. I expect those two tasks to be the most difficult ones in term of complexity (:crossed_fingers:). 

@osanseviero I'd say most of the PR is now reviewable and especially the two markdown pages. However, content is still a bit empty given that parameters are rarely documented [in the specs](https://github.com/huggingface/huggingface.js/tree/main/packages/tasks/src/tasks/chat-completion/spec). We will have to fix this in a separate PR in [huggingface.js](https://github.com/huggingface/huggingface.js). I do think we shouldn't wait for specs update to merge this PR.

### CODE REVIEW

**Comment by Wauplin** [comment]
Adding an example of the rendered table

image

File: docs/api-inference/tasks/chat_completion.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+This is a subtask of [`text-generation`](./text_generation) designed to generate responses in a conversational context.
+
+
+
+### Recommended models
+
+- [google/gemma-2-2b-it](https://huggingface.co/google/gemma-2-2b-it): A text-generation model trained to follow instructions.
+- [meta-llama/Meta-Llama-3.1-8B-Instruct](https://huggingface.co/meta-llama/Meta-Llama-3.1-8B-Instruct): Very powerful text generation model trained to follow instructions.
+- [microsoft/Phi-3-mini-4k-instruct](https://huggingface.co/microsoft/Phi-3-mini-4k-instruct): Small yet powerful text generation model.
+- [AI-MO/NuminaMath-7B-TIR](https://huggingface.co/AI-MO/NuminaMath-7B-TIR): A very powerful model that can solve mathematical problems.
```
Comment (osanseviero) [inline docs/api-inference/tasks/chat_completion.md:13]: As discussed in a follow-up, we should prevent having frozen models here

File: docs/api-inference/tasks/chat_completion.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+API_URL = "https://api-inference.huggingface.co/models/undefined"
+headers = {"Authorization": "Bearer hf_***"}
+
+def query(payload):
+	response = requests.post(API_URL, headers=headers, json=payload)
+	return response.json()
+	
+output = query({
+	"inputs": "Can you please let us know more details about your ",
```
Comment (osanseviero) [inline docs/api-inference/tasks/chat_completion.md:168]: To update to conversational API before release

File: docs/api-inference/tasks/text_generation.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
+## Text generation
+
+Generate text based on a prompt.
+
+<Tip>
+
+For more details about the `text-generation` task, check out its [dedicated page](https://huggingface.co/tasks/text-generation)! You will find examples and related materials.
```
Comment (osanseviero) [inline docs/api-inference/tasks/text_generation.md:7]: We should add some text that refers to chat_completion

**Review by osanseviero — APPROVED** [review]
Looking nice!

File: docs/api-inference/tasks/chat_completion.md (hunk 1 above)
Comment (Wauplin) [reply docs/api-inference/tasks/chat_completion.md:13]: yes, related thread: https://github.com/huggingface/hub-docs/pull/1384#discussion_r1729042102

File: docs/api-inference/tasks/text_generation.md (hunk 3 above)
Comment (Wauplin) [reply docs/api-inference/tasks/text_generation.md:9]: addressed in https://github.com/huggingface/hub-docs/pull/1393/commits/3eb85e8082a187a9c998d50fcaaff29df305b160

File: docs/api-inference/tasks/chat_completion.md (hunk 2 above)
Comment (Wauplin) [reply docs/api-inference/tasks/chat_completion.md]: that was a mistake on my side. Fixed in https://github.com/huggingface/hub-docs/pull/1393/commits/486e809e12669c4c8c6152d7a0040927d0385425

File: docs/api-inference/tasks/chat_completion.md (hunk 1 above)
Comment (Wauplin) [reply docs/api-inference/tasks/chat_completion.md:13]: https://github.com/huggingface/hub-docs/pull/1396/files

**Pull request merged**